docs/docs/contribute/code/building-chainblocks.md

```diff
@@ -5,54 +5,123 @@ license: CC-BY-SA-4.0
+
+2. Make sure you pull this project repository with dependencies by using the following command (from any terminal): 
+    
+    `git submodule update --init --recursive`
+
+3. Ensure you're using the Rust GNU toolchain by firing the following rustup command (from any terminal):
+
+    `rustup default stable-x86_64-pc-windows-gnu`
+
+    ![Update User Acc Env Var on Windows](assets/build-cb_use-gnutoolchain.png)
+
+4. For windows, the path of MinGW bin should be added to your system environment PATH variable (Settings > Edit environment variables for your account > User variables for 'user' > Path). This is needed in order to run MinGW from powershell, vscode, etc. The value of this PATH is usually `C:\msys64\mingw64\bin` .
+
+![Update User Acc Env Var on Windows](assets/build-cb_acc-env-var.png)
+
+### Update system packages
 
-### Instructions
+Open mingw prompt (use `mingw64.exe` from msys2 folder).
 
-*We use gcc and clang a lot, MSVC might work but it's uncharted territory.*
+![Update User Acc Env Var on Windows](assets/build-cb_run-mingw64.png)
 
-Make sure you pull this repository with dependencies `git submodule update --init --recursive`
+And run the following commands to update system packages:
 
-Make sure to use the gnu toolchain for rust with `rustup default stable-x86_64-pc-windows-gnu`
+1. This command will do a full update. You might have to reopen the prompt window
+    ```
+    pacman -Syu --noconfirm
+    ``` 
+    ![Update User Acc Env Var on Windows](assets/build-cb_upd-pckg-1.png)
 
-1. Open mingw prompt (use `mingw64.exe` from msys2 folder), run those commands:
-    1. `pacman -Syu --noconfirm` - to do a full update, you might have to reopen the prompt window
-    2. `pacman -S --needed --noconfirm base-devel mingw-w64-x86_64-toolchain mingw-w64-x86_64-cmake mingw-w64-x86_64-boost mingw-w64-x86_64-ninja mingw-w64-x86_64-clang mingw-w64-x86_64-lld` - to install dependencies
-2. Add mingw bin to your system environment PATH variable (the windows one, control panel), in otder to run from powershell, vscode, etc
-    * Usually `C:\msys64\mingw64\bin`
-3. Open your favorite terminal (could be within visual studio code) pointing to this repo folder
-4. Run those commands:
+2. This command will install dependencies:
+    ```
+    pacman -S --needed --noconfirm base-devel mingw-w64-x86_64-toolchain mingw-w64-x86_64-cmake mingw-w64-x86_64-boost mingw-w64-x86_64-ninja mingw-w64-x86_64-clang mingw-w64-x86_64-lld
     ```
+    ![Update User Acc Env Var on Windows](assets/build-cb_upd-pckg-21.png)
+
+    ![Update User Acc Env Var on Windows](assets/build-cb_upd-pckg-22.png)
+
+### Build & run the project
+
+Open your favorite terminal (could be within Visual Studio Code) pointing to this repo folder and run the following sequence of commands.
+
+Create a build directory,
+    
     mkdir build
```

Comment:
Although 4 spaces is also detected as a code block,I'd rather we use code fences for consistency:

    ```
    mkdir build
    ```